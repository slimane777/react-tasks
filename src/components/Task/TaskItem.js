import React from 'react'
import PropTypes from 'prop-types'
import { TaskRow, TaskCheckBox, TaskLabel, TaskTime, CheckboxContainer, HiddenCheckbox, StyledCheckbox, Icon } from './styles'
import { useDispatch } from 'react-redux'
import moment from 'moment'
import {
  UPDATE_TASK_REQUESTED
} from '../../redux/actions/tasks'

const TaskItem = ({ task }) => {

  const dispatch = useDispatch()

  const updateTask = (taskObj) => dispatch({ type: UPDATE_TASK_REQUESTED, payload: taskObj });

  return (
    // <TaskRow onClick={()=>updateTask({id: task.id, title: task.title, completed: !task.completed, dateOfComplete: moment(new Date())})}>
    <TaskRow 
      onClick={(e)=>{
        console.log({e})
        console.log({task})
        e.preventDefault()
        let obj = {
          id: task.id, 
          title: task.title, 
          completed: !task.completed, 
          dateOfComplete: moment(new Date())
        }
        console.log({obj})
        updateTask(obj)
      }}
    >
      {/* <TaskCheckBox 
          type="checkbox"
          value={task.title}
          checked={task.completed}
          onChange={()=>dispatch()}
      /> */}

      <CheckboxContainer >
        {/* <HiddenCheckbox checked={task.completed} /> */}
        <StyledCheckbox 
          checked={task.completed ? task.completed : false} 
          // onChange={(e)=>{
          //   console.log({e})
          //   // let obj = {
          //   //   id: task.id, 
          //   //   title: task.title, 
          //   //   completed: !task.completed, 
          //   //   dateOfComplete: moment(new Date())
          //   // }
          //   // updateTask(obj)
          // }}
        >
          {
            task.completed &&
            <Icon viewBox="0 0 24 24">
              <polyline points="20 6 9 17 4 12" />
            </Icon>
          }
        </StyledCheckbox>
      </CheckboxContainer>

      <TaskLabel checked={task.completed}>
        {task.title}
      </TaskLabel>
      <TaskTime>
        {moment(new Date()).format('h:MM a')}
      </TaskTime>
     </TaskRow>
  )
}

TaskItem.propTypes = {
  task: PropTypes.object.isRequired,
}

export default TaskItem