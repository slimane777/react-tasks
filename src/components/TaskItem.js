import React from 'react'
import PropTypes from 'prop-types'

const TaskItem = ({ task, deleteTask }) => {
  return (
    <p 
      onDoubleClick={() => deleteTask(task.id)} 
      style={{ cursor: 'pointer' }}
     >
       {task.title}
     </p>
  )
}

TaskItem.propTypes = {
  task: PropTypes.object.isRequired,
  deleteTask: PropTypes.func.isRequired
}

export default TaskItem