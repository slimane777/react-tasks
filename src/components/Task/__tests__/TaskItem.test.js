import React from 'react';
import renderer from 'react-test-renderer';
import TaskItem from '../TaskItem';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

describe('TaskItem test', () => {
  const initialState = {tasks:{tasks: []}}
  const mockStore = configureStore()
  let store,task

  it('TaskItem should match snapshot', () => {
    store = mockStore(initialState)
    task = {}
    const component = renderer.create(<Provider store={store}><TaskItem
      task={task} /></Provider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
