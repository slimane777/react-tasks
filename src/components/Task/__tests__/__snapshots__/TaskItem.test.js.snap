// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskItem test TaskItem should match snapshot 1`] = `
<div
  className="sc-eCApnc eqSazG"
  onClick={[Function]}
>
  <div
    className="sc-gKAaRy kioxeA"
  >
    <div
      checked={false}
      className="sc-jSFjdj egBuVt"
    />
  </div>
  <p
    checked={false}
    className="sc-fujyAs clfZMC"
  />
  <span
    className="sc-pNWdM fNELWa"
  >
    4:07 pm
  </span>
</div>
`;
